src/overlay/TCPPeer.cpp

```diff
@@ -440,7 +440,12 @@ TCPPeer::startRead()
     // We read large-ish (256KB) buffers of data from TCP which might have quite
     // a few messages in them. We want to digest as many of these
     // _synchronously_ as we can before we issue an async_read against ASIO.
-    YieldTimer yt(mApp.getClock(), mApp.getConfig().MAX_BATCH_READ_PERIOD_MS,
+    //
+    // This loop is charged to the IO debt-tracker as it effectively represents
+    // time spent "draining events" from the asio::io_context, though they're
+    // just input-buffer-consumption steps happening in userspace.
+    YieldTimer yt(mApp.getClock(), mApp.getClock().getIODebtTracker(),
```

Comment:
you're double counting work, `startRead` can be called from :
1. the execution queue, in which case you're going to count the execution time against both trackers.There is a risk of causing both trackers to stall. Maybe it would be better to deduce this from the execution timer (this can be done by recording the amount of debt before invoking the execution loop, and crediting any change back to the execution loop)?
2. async callbacks, here it's double counting time against iotracker
   * the simplest thing to do here is to set a flag in the tracker as "in use" from `crank` as to not track debt here.
   * you could do something else so that you can have this loop break out early if "you ran out of time" in the outer yield timer, but that doesn't seem super useful